demonstrations/tutorial_expressivity_fourier_series.py

```diff
@@ -0,0 +1,800 @@
+# We now define the quantum model itself.
+# 
+
+scaling = 1
+
+dev = qml.device('default.qubit', wires=1)
+
+def S(x):
+    """Data encoding circuit block."""
+    qml.RX(scaling*x, wires=0)
+
+def W(theta):
+    """Trainable circuit block."""
+    qml.Rot(theta[0], theta[1], theta[2], wires=0)
+
+    
+@qml.qnode(dev)
+def serial_quantum_model(weights, x=None):
+    
+    for theta in weights[:-1]:
+        W(theta)
+        S(x)
+        
+    # L+1'th unitary
+    W(weights[-1])
+    
+    return qml.expval(qml.PauliZ(wires=0))
+
+
+######################################################################
+# You can run the following cell multiple times, each time sampling
+# different weights, and therefore different quantum models.
+# 
+
+r = 1 # number of times the encoding gets repeated (here equal to the number of layers)
+weights = 2*np.pi*np.random.random(size=(r+1, 3)) # some random initial weights
+
+x = np.linspace(-6, 6, 70)
+random_quantum_model_y = [serial_quantum_model(weights, x=x_) for x_ in x]
+
+plt.plot(x, random_quantum_model_y, c='blue')
+plt.scatter(x, random_quantum_model_y, c='blue', s=10)
+plt.ylim(-1,1)
+plt.show()
+
+
+######################################################################
+# 
+# No matter what weights are picked, the single qubit model for `L=1` will always be a sine function 
+# of a fixed frequency. The weights merely influence the amplitude, y-shift and phase of the sine.
+# 
+# This observation is formally derived in Section II.A of the paper.
+# 
+
+
+######################################################################
+# .. note:: 
+# 
+#     You can increase the number of layers. Figure 4 from the paper, for
+#     example, uses the settings ``L=1``, ``L=3`` and ``L=5``.
+# 
+
+
+######################################################################
+# Finally, let's look at the circuit we just created:
+# 
+
+print(serial_quantum_model.draw())
+
+
+######################################################################
+# Fit the model to the target
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# 
+
+
+######################################################################
+# The next step is to optimise the weights in order to fit the ground
+# truth.
+# 
+
+def cost(weights, x, y):
+    predictions = [serial_quantum_model(weights, x=x_) for x_ in x]
+    return square_loss(y, predictions)
+
+max_steps = 50
+opt = qml.AdamOptimizer(0.3)
+batch_size = 25
+cst = [cost(weights, x, target_y)]# initial cost
+
+for step in range(max_steps):
+
+    # Select batch of data
+    batch_index = np.random.randint(0, len(x), (batch_size,))
+    x_batch = x[batch_index]
+    y_batch = target_y[batch_index]
+
+    # Update the weights by one optimizer step
+    weights = opt.step(lambda w: cost(w, x_batch, y_batch), weights)
+
+    # Save, and possibly print, the current cost
+    c = cost(weights, x, target_y)
+    cst.append(c)
+    if step % 10 == 0:
```

Comment:
```suggestion
    if (step + 1) % 10 == 0:
```
step == 0 is actually the first step. In this way step 50 will also get printed